### PR TITLE
Added more customizations to pulsebar and alsabar and light widget

### PR DESCRIPTION
**Pull request by mtxr**
I've opened this PR before, but my buttons setup were in the code.

I'll be updating the wiki right now.

I think that would be great to exports commands as well to help to integrate with buttons because it's easier to manage the command in just one place

### CODE REVIEW

**Comment by mtxr** [comment]
Well, just saw you already updated the wiki before.
Thanks. Awesome project btw.

**Comment by mtxr** [comment]
https://github.com/lcpz/lain/wiki/light
The wiki for light widget

File: widget/alsabar.lua [hunk 1]
```diff
@@ -28,18 +28,22 @@ local function factory(args)
...
     local settings   = args.settings or function() end
     local width      = args.width or 63
     local height     = args.height or 1
+    local margins    = args.margins or 1
+    local paddings   = args.paddings or 1
     local ticks      = args.ticks or false
     local ticks_size = args.ticks_size or 7
+    local step       = args.step or '5%'
 
-    alsabar.cmd                 = args.cmd or "amixer"
```
Comment (lcpz) [inline widget/alsabar.lua]: This is inconsistent and breaks the API.

File: widget/alsabar.lua [hunk 2]
```diff
@@ -51,20 +55,23 @@ local function factory(args)
         alsabar.notification_preset.font = "Monospace 10"
     end
 
-    local format_cmd = string.format("%s get %s", alsabar.cmd, alsabar.channel)
+    alsabar.cmds.get_cmd = string.format("%s get %s", alsabar.alsa_bin, alsabar.channel)
+    alsabar.cmds.inc_cmd = string.format("%s sset %s %s+", alsabar.alsa_bin, alsabar.channel, step)
+    alsabar.cmds.dec_cmd = string.format("%s sset %s %s-" , alsabar.alsa_bin, alsabar.channel, step)
+    alsabar.cmds.tog_cmd = string.format("%s sset %s toggle", alsabar.alsa_bin, alsabar.channel)
```
Comment (lcpz) [inline widget/alsabar.lua]: These commands are just useless here if the user does not need them.

But we can add them to a theming/configuring submodule. https://github.com/lcpz/lain/issues/317

File: widget/alsabar.lua [hunk 3]
```diff
@@ -51,20 +55,23 @@ local function factory(args)
...
     alsabar.bar = wibox.widget {
         forced_height    = height,
         forced_width     = width,
         color            = alsabar.colors.unmute,
         background_color = alsabar.colors.background,
-        margins          = 1,
-        paddings         = 1,
+        margins          = margins,
+        paddings         = paddings,
         ticks            = ticks,
```
Comment (lcpz) [inline widget/alsabar.lua]: Already present in latest master commit. Please update your clone.

**Comment by lcpz** [comment]
I did not see [7eece9b](https://github.com/lcpz/lain/pull/406/commits/7eece9bbed3a800eb2c9fbb30ffb164328bf2901) before commenting on [d9c46a5](https://github.com/lcpz/lain/pull/406/commits/d9c46a5423cb14708e54757a1489ec3435cd256e), but you gave me an idea (see comments). By the way, if you have any suggestions, feel free to add them in https://github.com/lcpz/lain/issues/317.

Regarding the new widget, see if you can create a [`watch` recipe](https://awesomewm.org/recipes/watch) instead. We will put it on the Awesome website. Otherwise, please sync with lain master and open a new PR with just the commit related to the new code. Note that contributed widgets have to be put in `widgets/contrib`.

**Pull request closed without merging**